packages/@aws-cdk/aws-synthetics/README.md

```diff
@@ -1,16 +1,142 @@
+```
+
+> **Note:** The function **must** be called `handler`.
+
+The canary will automatically produce a CloudWatch Dashboard:
+
+![UI Screenshot](images/ui-screenshot.png)
+
+The Canary code will be executed in a lambda function created by Synthetics on your behalf. The Lambda function includes a custom [runtime](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_Library.html) provided by Synthetics. The provided runtime includes a variety of handy tools such as [Puppeteer](https://www.npmjs.com/package/puppeteer-core)  and Chromium. To learn more about Synthetics capabilities, check out the [docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries.html)
+
+### Configuring the Canary Script 
+
+To configure the script the canary executes, use the `test` property. The `test` property accepts a `Test` instance that can be initialized by the `Test` class static methods. Currently, the only implemented method is `Test.custom()`, which allows you to bring your own code. In the future, other methods will be added. `Test.custom()` accepts `code` and `handler` properties -- both are required by Synthetics to create a lambda function on your behalf.
+
+> **Note:** For `code.fromAsset()` and `code.fromBucket()`, the canary resource requires the following folder structure:
+>```
+>canary/
+>├── nodejs/
+>   ├── node_modules/
+>        ├── <filename>.js
+>```
+> See Synthetics [docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_WritingCanary.html).
+
+The `synthetics.Code` class exposes static methods to bundle your code artifacts: 
+
+  - `code.fromInline(code)` - specify an inline script.
+  - `code.fromAsset(path)` - specify a .zip file or a directory in the local filesystem which will be zipped and uploaded to S3 on deployment. See the above Note for directory structure.
+  - `code.fromBucket(bucket, key[, objectVersion])` - specify an S3 object that contains the .zip file of your runtime code. See the above Note for directory structure.
+
+```ts
+// To supply the code inline:
+const canary = new Canary(this, 'MyCanary', {
+  test: Test.custom({
+    code: Code.fromInline('/* Synthetics handler code */'),
+    handler: 'index.handler', // must be 'index.handler'
+  }),
+});
+
+// To supply the code from your local filesystem:
+const canary = new Canary(this, 'MyCanary', {
+  test: Test.custom({
+    code: Code.fromAsset(path.join(__dirname, 'canary')),
+    handler: 'index.handler', // must end with '.handler'
+  }),
+});
+
+// To supply the code from a S3 bucket:
+const canary = new Canary(this, 'MyCanary', {
+  test: Test.custom({
+    code: Code.fromBucket(bucket, 'canary.zip'),
+    handler: 'index.handler', // must end with '.handler'
+  }),
+}); 
+```
+
+### Alarms
+
+You can configure a CloudWatch Alarm on canary metrics. Metrics are emitted by CloudWatch automatically and can be accessed by the following APIs:
```

Comment:
```suggestion
You can configure a CloudWatch Alarm on the canary metrics. Metrics are emitted by CloudWatch automatically and can be accessed by the following APIs:
```

Reply:
I changed it to "You can configure a CloudWatch Alarm on a canary metric"